src/SamplesApp/SamplesApp.UITests/Microsoft_UI_Xaml_Controls/NumberBoxTests/Given_NumberBox.cs

```diff
@@ -60,35 +59,12 @@ public void UpDownTest()
 			Assert.AreEqual(100, numBox.GetDependencyPropertyValue<double>("Value"));
 
 			Console.WriteLine("Assert that incrementing after typing in a value validates the text first.");
-			numBox.ClearText();
-			numBox.EnterText("50");
+			EnterTextInNumberBox(numBox, "50");
 			_app.PressEnter();
 			upButton.FastTap();
 			Assert.AreEqual(55, numBox.GetDependencyPropertyValue<double>("Value"));
 		}
 
-		[Test]
-		[AutoRetry]
-		public void DecimalFormatterTest()
-		{
-			Run("UITests.Shared.Microsoft_UI_Xaml_Controls.NumberBoxTests.NumberBoxPage");
-
-			var numBox = _app.Marked("TestNumberBox");
-			Assert.AreEqual(0, numBox.GetDependencyPropertyValue<double>("Value"));
-
-			_app.FastTap("MinCheckBox");
-			_app.FastTap("MaxCheckBox");
-			_app.Marked("CustomFormatterButton").FastTap();
-
-			numBox.ClearText();
-			numBox.EnterText("۱٫۷");
-			_app.PressEnter();
-			_app.PressEnter();
-
-			Assert.AreEqual("۱٫۷۵", numBox.GetDependencyPropertyValue<string>("Text"));
-			Assert.AreEqual(1.75, numBox.GetDependencyPropertyValue<double>("Value"));
```

Comment:
It's unclear to me why this is the correct behavior? We entered `1.7`, how it got into `1.75` suddenly?

Reply:
It depends on how IncrementNumberRounder works it's explained [here](https://docs.microsoft.com/en-us/uwp/api/windows.globalization.numberformatting.incrementnumberrounder.increment?view=winrt-22621#remarks)

Reply:
Thanks @MohammadHadi2031.

Looking at the NumberBox implementation, my understanding is that this is about the *display* (`Text`), not `Value`:

https://github.com/unoplatform/uno/blob/4ba74688801eade7325d4d1fdf034d75e3742f52/src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs#L537

How does this reflect into the `Value`? `OnTextPropertyChanged` shouldn't reflect into `Value` for this case because we set `m_textUpdating` to `true`.

What am I missing?

Reply:
I figured that out.

Once we enter 1.7, the text becomes `۱٫۷۵`, but the value remains 1.7.
The extra PressEnter in the test acts as if it's entering a new text which is `۱٫۷۵`, and now value becomes 1.75.